File: frontend/src/lib/preferences.ts
import { api } from "./api";

export interface Preferences {
  theme: "light" | "dark" | "system";
  editor_font_size: number;
  editor_minimap: boolean;
  editor_word_wrap: boolean;
}

export async function getPreferences(): Promise<Preferences> {
  return api.request<Preferences>("/users/me/preferences");
}

export async function updatePreferences(
  prefs: Preferences
): Promise<Preferences> {
  return api.request<Preferences>("/users/me/preferences", {
    method: "PUT",
    body: JSON.stringify(prefs),
  });
}

File: backend/app/schemas/preferences.py
from pydantic import BaseModel, Field


class PreferencesResponse(BaseModel):
    theme: str
    editor_font_size: int
    editor_minimap: bool
    editor_word_wrap: bool


class PreferencesUpdateRequest(BaseModel):
    theme: str = Field(pattern=r"^(light|dark|system)$")
    editor_font_size: int = Field(ge=8, le=32)
    editor_minimap: bool
    editor_word_wrap: bool
